package.json

```diff
@@ -20,7 +20,7 @@
     "lodash.get": "^4.4.2"
   },
   "peerDependencies": {
-    "graphql": "^0.13.0 || ^14.0.0"
```

Comment:
Do we have to drop support for 0.13 and 14.0?

Reply:
Sort of... The issue here is that `onError` was only added to the constructor args of `ValidationContext` in 14.5.0.

We don't strictly need that argument outside of tests, and we only pass a dummy function through anyway, so theoretically we still have runtime compatibility with `^0.13.0 || ^14.0.0`. But, there'd be a type error trying to build against those versions, and the tests wouldn't work.

Reply:
Thanks for merging the CI PR - I've just pulled that into this branch to demonstrate the problem here - as you can see the 0.13 and 14.0 tests fail, **but** this is only because the tests themselves now rely on the new `onError` API instead of the previous `getErrors`.

It would be possible to fix this by adding a compatibility shim for the use within tests only, which will work with any version of graphql:

```ts
class CompatibleValidationContext extends ValidationContext {
  private errors: GraphQLError[] = []

  constructor(schema: GraphQLSchema, ast: DocumentNode, typeInfo: TypeInfo) {
    super(schema, ast, typeInfo, err => this.errors.push(err));
  }

  getErrors(): ReadonlyArray<GraphQLError> {
    // @ts-ignore
    return super.getErrors ? super.getErrors() : this.errors
  }
}
```

So I see three options:

1. Add (something like) the compatibility layer above for use in tests, and keep CI and peer deps as they are.
2. Remove versions below 14.5 from CI, but keep them in peer dependencies. It'll work for now, and keeps tests tidy, but you'd have to keep an eye out for regressions.
3. Bump peer dependency up to >= 14.5

Your call sir!

Reply:
Thanks for the detailed info here! I'd say let's add this compatibility layer (just for the tests) for now with a little note to remove this once we drop support for older GraphQL versions. That should give people time to upgrade their codebases and once we drop support for the old versions, we can remove that temporary compatibility layer.

Reply:
Sounds good, done. I wasn't sure what to call `CompatibleValidationLayer` or exactly where to put it - let me know if you'd like any of that changing.

Otherwise, all tests now green, and I think we're done?